service-api/app/src/App/src/Handler/LpasActionsHandler.php

```
@@ -23,29 +25,58 @@ class LpasActionsHandler implements RequestHandlerInterface
+    private ActorCodeService $actorCodeService;
+
+    public function __construct(LpaService $lpaService, ActorCodeService $actorCodeService)
     {
         $this->lpaService = $lpaService;
+        $this->actorCodeService = $actorCodeService;
     }
 
     /**
      * @param ServerRequestInterface $request
      * @return ResponseInterface
+     * @throws \Exception
      */
     public function handle(ServerRequestInterface $request): ResponseInterface
     {
+        $lpaMatchResponse = null;
         $requestData = $request->getParsedBody();
+        $userId = $request->getAttribute('user-id');
+
+        if (
+            !isset($requestData['reference_number']) ||
+            !isset($requestData['dob']) ||
+            !isset($requestData['first_names']) ||
+            !isset($requestData['last_name']) ||
+            !isset($requestData['postcode'])
+        ) {
+            throw new BadRequestException("Required data missing!");
+        }
+        // Check LPA with user provided reference number
+        $lpaMatchResponse = $this->lpaService->checkLPAMatchAndGetActorDetails($requestData);
 
-        if (!isset($requestData['lpa-id'])) {
+        if (is_null($lpaMatchResponse)) {
+            throw new NotFoundException('LPA not found');
+        }
+
+        if (!isset($lpaMatchResponse['lpa-id'])) {
             throw new BadRequestException("'lpa-id' missing.");
         }
 
-        if (!isset($requestData['actor-id'])) {
+        if (!isset($lpaMatchResponse['actor-id'])) {
             throw new BadRequestException("'actor-id' missing.");
         }
 
-        $this->lpaService->requestAccessByLetter($requestData['lpa-id'], $requestData['actor-id']);
+        // Checks if the actor already has an active activation key is_null(abc):true
+        $hasActivationCode = $this->actorCodeService->hasActivationCode($requestData['reference_number'], $lpaMatchResponse['actor-id']);
+        if (!$hasActivationCode) {
```

@SeemaMenon The logic is backwards here, you'll need to remove the ! as when $hasActivationCode is true (they do have a valid code) you'll want to throw the exception